pkg/mimir/tracing.go

```diff
@@ -11,29 +11,13 @@ import (
 
 	"github.com/opentracing/opentracing-go"
 	"github.com/opentracing/opentracing-go/log"
-	"github.com/thanos-io/thanos/pkg/tracing"
 	"github.com/uber/jaeger-client-go"
 	"go.opentelemetry.io/otel/attribute"
 	"go.opentelemetry.io/otel/codes"
 	"go.opentelemetry.io/otel/trace"
 	"google.golang.org/grpc"
 )
 
-// ThanosTracerUnaryInterceptor injects the opentracing global tracer into the context
-// in order to get it picked up by Thanos components.
-func ThanosTracerUnaryInterceptor(ctx context.Context, req interface{}, _ *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (interface{}, error) {
-	return handler(tracing.ContextWithTracer(ctx, opentracing.GlobalTracer()), req)
```

Comment:
The fix in the objstore repo is minimal, so I've opened a PR:
https://github.com/thanos-io/objstore/pull/33